### PR TITLE
Add review_checklist.md

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Add reviewer_checklist.md

Can be used in the other collections where polished enough

### CODE REVIEW

**Comment by Andersson007** [comment]
@Jorge-Rodriguez , any ideas how to improve it?
Take your time. I'll suggest this list to use everywhere across the Ansible collections later.

File: review_checklist.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Review Checklist
+
+When reviewing, keep in mind that we follow [Ansible Code of Conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html) in all our contributions and interactions within this repository.
+
+**General tips**
+- Try create a culture of collaboration, when reviewing
```
Comment (aminvakil) [inline review_checklist.md]: I'm not sure about this change myself though, but I think "try create" is not correct.
```suggestion
- Try to create a culture of collaboration when reviewing
```

File: review_checklist.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- Welcome the author and thank them for the pull request
+- When suggesting changes, try to use questions, not statements
+- When suggesting mandatory changes, do it as politely as possible providing documentation references
+- If your suggestion is optional or a matter of personal preferences, please say it explicitly
+- When asking for adding tests or for complex code refactoring, say that the author is welcome to ask for clarifications and help if they need
+- If somebody suggests a good idea, say about it or put a thumbs up
+- After merging, thank the author and reviewers for their time and effort
+
+**Standards and documentation**
+- [ ] if the pull request is not a documentation fix, it must include a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#creating-a-changelog-fragment). Please check the format carefully
```
Comment (aminvakil) [inline review_checklist.md]:
```suggestion
- [ ] if the pull request is not a documentation fix, it must include a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#creating-a-changelog-fragment). Please check the format carefully.
```

File: review_checklist.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+- [ ] if new files are added with the pull request, they follow the [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
+- [ ] the changes follow the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
+- [ ] follow [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
+- [ ] if a new plugin is added, it is one of the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#id14)
```
Comment (aminvakil) [inline review_checklist.md]:
```suggestion
- [ ] if a new plugin is added, it is one of the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins)
```

File: review_checklist.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+- [ ] documentation, examples, and return sections use FQCNs for `M(..)` when referencing to modules
+- [ ] modules (or plugins) from ansible-core use `ansible.builtin.` as a FQCN prefix when mentioned
+- [ ] when a new option, module, plugin, or return value is added, the corresponding documentation or return sections use `version_added:` containing the *collection* version for which they were added
+- [ ] FQCNs is used for `extends_documentation_fragment:`, unless the author is referring to doc_fragments from ansible-core
+
+**Tests (if applicable and possible to implement):** 
+- [ ] the pull request has integration tests 
+- [ ] the pull request has unit tests
+- [ ] all changes are covered
+- [ ] integration tests also cover `check_mode`
```
Comment (aminvakil) [inline review_checklist.md]:
```suggestion
- [ ] integration tests also cover `check_mode` (if it is supported)
```

File: review_checklist.md (hunk 1 above)
Comment (Andersson007) [reply review_checklist.md]: Good spot, thanks! committed

File: review_checklist.md (hunk 2 above)
Comment (Andersson007) [reply review_checklist.md]: yes, i know about it. If I add the dot, not sure if the next sentence should start with a capital letter and so on.. let's try another approach

Comment (Andersson007) [reply review_checklist.md]: PTAL

File: review_checklist.md (hunk 4 above)
Comment (Andersson007) [reply review_checklist.md]: well spotted, committed:)

File: REVIEW_CHECKLIST.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+When reviewing, keep in mind that we follow [Ansible Code of Conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html) in all our contributions and interactions within this repository.
+
+**General tips**
+- Try to create a culture of collaboration when reviewing
+- Welcome the author and thank them for the pull request
+- When suggesting changes, try to use questions, not statements
+- When suggesting mandatory changes, do it as politely as possible providing documentation references
+- If your suggestion is optional or a matter of personal preferences, please say it explicitly
+- When asking for adding tests or for complex code refactoring, say that the author is welcome to ask for clarifications and help if they need
+- If somebody suggests a good idea, say about it or put a thumbs up
```
Comment (aminvakil) [inline REVIEW_CHECKLIST.md]: Please feel free to ignore this as I'm really not sure about this myself.
```suggestion
- If somebody suggests a good idea, mention it or put a thumbs up
```

File: REVIEW_CHECKLIST.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+- [ ] modules and plugins from ansible-core use `ansible.builtin.` as a FQCN prefix when mentioned
+- [ ] when a new option, module, plugin, or return value is added, the corresponding documentation or return sections use `version_added:` containing the *collection* version for which they were added
+- [ ] FQCNs is used for `extends_documentation_fragment:`, unless the author is referring to doc_fragments from ansible-core
+
+**Tests (if applicable and technically possible to implement):**
+- [ ] the pull request has integration tests 
+- [ ] the pull request has unit tests
+- [ ] all changes are covered
+- [ ] integration tests also cover `check_mode` (if it is supported)
+- [ ] integration tests check an actual state of the system, not only what the module reports (for example, if the module change a file, check that the file was actually changed by using the `ansible.builtin.stat` module)
```
Comment (aminvakil) [inline REVIEW_CHECKLIST.md]:
```suggestion
- [ ] integration tests check an actual state of the system, not only what the module reports (for example, if the module changes a file, check that the file was actually changed by using the `ansible.builtin.stat` module)
```

File: review_checklist.md (hunk 2 above)
Comment (aminvakil) [reply review_checklist.md]: Thanks for the change, it's more convenient now imho.

Comment (Andersson007) [reply review_checklist.md]: @aminvakil thanks for spotting this and giving the feedback

File: REVIEW_CHECKLIST.md (hunk 5 above)
Comment (Andersson007) [reply REVIEW_CHECKLIST.md]: sounds better, thanks! committed

File: REVIEW_CHECKLIST.md (hunk 6 above)
Comment (Andersson007) [reply REVIEW_CHECKLIST.md]: good eye:)

**Comment by Andersson007** [comment]
added refs for integration and unit tests

File: REVIEW_CHECKLIST.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+**Standards and documentation**
+- [ ] if the pull request is not a documentation fix, it must include a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#creating-a-changelog-fragment) - please check the format carefully
+- [ ] if new files are added with the pull request, they follow the [licensing rules](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#licensing)
+- [ ] the changes follow the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
+- [ ] the changes follow the [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html)
+- [ ] if a new plugin is added, it is one of the [allowed plugin types](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules-plugins)
+- [ ] documentation, examples, and return sections use FQCNs for `M(..)` when referring to modules
```
Comment (Jorge-Rodriguez) [inline REVIEW_CHECKLIST.md]: A link here to relevant documentation or examples would be welcome.

File: REVIEW_CHECKLIST.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+- [ ] modules and plugins from ansible-core use `ansible.builtin.` as a FQCN prefix when mentioned
+- [ ] when a new option, module, plugin, or return value is added, the corresponding documentation or return sections use `version_added:` containing the *collection* version for which they were added
```
Comment (Jorge-Rodriguez) [inline REVIEW_CHECKLIST.md]: I always got this wrong. Wasn't there some sort of key version numbers that the release automation picked up?

File: REVIEW_CHECKLIST.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+- [ ] FQCNs is used for `extends_documentation_fragment:`, unless the author is referring to doc_fragments from ansible-core
```
Comment (Jorge-Rodriguez) [inline REVIEW_CHECKLIST.md]:
```suggestion
- [ ] FQCNs are used for `extends_documentation_fragment:`, unless the author is referring to doc_fragments from ansible-core
```

**Review by Jorge-Rodriguez — COMMENTED** [review]
Some of the checklist items, e.g. unit tests, are not strictly mandatory. It could be a good idea to include a note to use appropriate judgement when going through the list.

File: REVIEW_CHECKLIST.md (hunk 9 above)
Comment (Andersson007) [reply REVIEW_CHECKLIST.md]: good eye:) committed, thanks

File: REVIEW_CHECKLIST.md (hunk 7 above)
Comment (Andersson007) [reply REVIEW_CHECKLIST.md]: Added, thanks!

File: REVIEW_CHECKLIST.md (hunk 8 above)
Comment (Andersson007) [reply REVIEW_CHECKLIST.md]: Added clarifications, thank Felix Fontein.
Is it now better?

**Review by Jorge-Rodriguez — APPROVED** [review]
LGTM

**Comment by Andersson007** [comment]
@aminvakil @Jorge-Rodriguez thanks for reviewing!

**Pull request merged**